test/Makefile

```diff
@@ -9,37 +9,43 @@
 # - `run`: execute the benchmarks with a Wasm `$(ENGINE)` of choice (e.g.,
 #   Wasmtime)
 
-test: run run_smoke
-
-# Unlike the `libc-test` directory, we will output all artifacts to the `build`
-# directory (keeping with the `wasi-libc` conventions).
-OBJDIR ?= $(CURDIR)/build
-DOWNDIR ?= $(CURDIR)/download
+test: run
 
+# Decide which target to build for and which libc to use.
 TARGET_TRIPLE ?= wasm32-wasi
 
+# Setup various paths used by the tests.
+OBJDIR ?= build/$(TARGET_TRIPLE)
+DOWNDIR ?= build/download
+SRCDIR ?= src
+RUNDIR ?= run/$(TARGET_TRIPLE)
+
+# We also need to know the location the wasi-libc sysroot we're building
+# against.
+SYSROOT_DIR ?= ../sysroot
+SYSROOT := $(SYSROOT_DIR)/lib/$(TARGET_TRIPLE)
+$(SYSROOT):
+	@echo "No sysroot for $(TARGET_TRIPLE) available at $(SYSROOT_DIR); to build it, e.g.:"
+	@echo "  cd $(dir $(SYSROOT_DIR))"
+	@echo "  make TARGET_TRIPLE=$(TARGET_TRIPLE)"
+	@exit 1
+
+
 ##### DOWNLOAD #################################################################
 
 LIBC_TEST_URL ?= https://github.com/bytecodealliance/libc-test
 LIBC_TEST = $(DOWNDIR)/libc-test
 LIBRT_URL ?= https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-24/libclang_rt.builtins-wasm32-wasi-24.0.tar.gz
 LIBRT = $(DOWNDIR)/libclang_rt.builtins-wasm32.a
 WASMTIME_URL ?= https://github.com/bytecodealliance/wasmtime/releases/download/v26.0.1/wasmtime-v26.0.1-x86_64-linux.tar.xz
-WASMTIME = $(DOWNDIR)/$(shell basename $(WASMTIME_URL) .tar.xz)/wasmtime
+WASMTIME = $(abspath $(DOWNDIR)/$(shell basename $(WASMTIME_URL) .tar.xz)/wasmtime)
```

Comment:
Huh, TIL `basename` took and optional suffix

Reply:
😁